src/test/java/com/google/alpollo/DataBaseTest.java

```diff
@@ -0,0 +1,121 @@
+  private static final int TEN = 10;
+  /** 
+   * We save 10 songs to database and this constant provides us to check adding more then 10 songs, 
+   * in this case 15, but we can change. 
+   */
+  private static final int MORE_THAN_TEN = 15;
+
+  @BeforeClass
+  public static void beforeClass() throws Exception {
+    ObjectifyService.setFactory(new ObjectifyFactory());
+    ObjectifyService.register(SongCounter.class);
+    ObjectifyService.register(AnalysisInfo.class);
+  }
+
+  @Before
+  public void before() throws Exception {
+    this.session = ObjectifyService.begin();
+    this.helper.setUp();
+  }
+
+  @After
+  public void after() {
+    AsyncCacheFilter.complete();
+    this.session.close();
+    this.helper.tearDown();
+  }
+
+  /** Check that at the begining database is empty. */
+  @Test
+  public void noRequests() {
+    Assert.assertEquals(new ArrayList<SongCounter>(), SongDataBase.topSongs());
+  }
+
+  /** Compare songs from database with correct list. */
+  private void checkList(List<Song> exceptedSongs) {
+    ArrayList<Song> actualSongs = new ArrayList<>();
+    SongDataBase.topSongs().forEach(songCounter -> actualSongs.add(songCounter.getSong()));
+    Assert.assertEquals(exceptedSongs.retainAll(actualSongs), actualSongs.retainAll(exceptedSongs));
+    Assert.assertTrue(exceptedSongs.size() <= TEN);
+  }
+
+  /** Add one song request and check that database returns list with this request. */
+  @Test
+  public void addOneRequest() {
+    SongDataBase.saveSongRequest(SONG);
+    ArrayList<Song> exceptedSongs = new ArrayList<>();
+    exceptedSongs.add(SONG);
+    checkList(exceptedSongs);
+  }
+
+  /** Add 10 requests and check that database returns list with them. */
+  @Test
+  public void addTenRequests() {
+    ArrayList<Song> exceptedSongs = new ArrayList<>(); 
+    for (int i = 0; i < TEN; i++) {
+      Song song = new Song(Integer.toString(i), Integer.toString(i + TEN));
+      exceptedSongs.add(song);
+      SongDataBase.saveSongRequest(song);
+    }
+    checkList(exceptedSongs);
+  }
+
+  /** Add more than 10 requests and check that database returns only 10 most searched. */
+  @Test
+  public void addMoreThanTenRequests() {
+    ArrayList<Song> exceptedSongs = new ArrayList<>(); 
+    for (int i = 0; i < MORE_THAN_TEN; i++) {
+      Song song = new Song(Integer.toString(i), Integer.toString(i + MORE_THAN_TEN));
+      for (int j = 0; j <= i; j++) {
+        SongDataBase.saveSongRequest(song);
+      }
+      // because last 10 songs will be serched more
+      if (MORE_THAN_TEN - TEN <= i) {
+        exceptedSongs.add(song);
+      }
+    }
+    checkList(exceptedSongs);
+  }
+
+  /** Serach for one song twice, for other once and check that fisrts song is in top10. */
+  @Test
+  public void changeOneCounter() {
+    SongDataBase.saveSongRequest(SONG);
+    for (int i = 0; i < TEN; i++) {
+      Song song = new Song(Integer.toString(i), Integer.toString(i + TEN));
```

Comment:
why you have `i` and `i + TEN`? if you want just to fill db with some different values you can just make it like `Song song = new Song(Integer.toString(i), Integer.toString(i));`. So it'll be more obvious that this object don't have some special meaning